### PR TITLE
support angular 16 in typescript-angular generator

### PR DESCRIPTION
**Pull request by twatzl**
this PR adds support for angular 16 by updating relevant package versions in the code generating package.json.

- rxjs
- typescript
- angular
- ngPackagr

this resolves the following issue https://github.com/OpenAPITools/openapi-generator/issues/15874

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

### CODE REVIEW

**Review by amakhrov — APPROVED** [review]
Thanks for the PR! All looks good for me.

Note that there is another similar PR (opened just a day ago): https://github.com/OpenAPITools/openapi-generator/pull/15925

The only substantial difference between the two, as I can see, is the required zone.js version for Angular 16.

**Comment by twatzl** [comment]
I see. We implemented the change last week, but did not manage to open the PR until yesterday. 
Anyway I don't mind which one is merged as long as everything works.

Regarding the zone.js version: might be the older one works as well. With the upgrade to angular 16 we upgraded all our frontends to zonejs 13 as well. In theory according to the docs no zonejs would be needed with angular 16 at all, but I assume that would require migration to signals.

To sum it up: I have no strong preferences here.

**Comment by akehir** [comment]
zone.js should probably be at 0.13; although in this context I'm not sure why it's necessary.

**Comment by macjohnny** [comment]
superseded by https://github.com/OpenAPITools/openapi-generator/pull/15925
